refactor: decoder state setup

Funatiq opened the pull request:
## Description

Please see commit messages.

## Test Coverage

<!--
Please list clearly what are the relevant test(s) that can safeguard the changes in the PR. This helps us to ensure we have sufficient test coverage for the PR.
-->

## GitHub Bot Help

`/bot [-h] ['run', 'kill', 'skip', 'reuse-pipeline'] ...`

Provide a user friendly way for developers to interact with a Jenkins server.

Run `/bot [-h|--help]` to print this help message.

See details below for each supported subcommand.

<details>

`run  [--disable-fail-fast --skip-test --stage-list "A10-1, xxx" --gpu-type "A30, H100_PCIe" --add-multi-gpu-test --only-multi-gpu-test --disable-multi-gpu-test --post-merge --extra-stage "H100_PCIe-[Post-Merge]-1, xxx"]`

Launch build/test pipelines. All previously running jobs will be killed.

`--disable-fail-fast ` *(OPTIONAL)* : Disable fail fast on build/tests/infra failures.

`--skip-test ` *(OPTIONAL)* : Skip all test stages, but still run build stages, package stages and sanity check stages. Note: Does **NOT** update GitHub check status.

`--stage-list "A10-1, xxx"` *(OPTIONAL)* : Only run the specified test stages. Examples: "A10-1, xxx". Note: Does **NOT** update GitHub check status.

`--gpu-type "A30, H100_PCIe"` *(OPTIONAL)* : Only run the test stages on the specified GPU types. Examples: "A30, H100_PCIe". Note: Does **NOT** update GitHub check status.

`--only-multi-gpu-test ` *(OPTIONAL)* : Only run the multi-GPU tests. Note: Does **NOT** update GitHub check status.

`--disable-multi-gpu-test ` *(OPTIONAL)* : Disable the multi-GPU tests. Note: Does **NOT** update GitHub check status.

`--add-multi-gpu-test ` *(OPTIONAL)* : Force run the multi-GPU tests. Will also run L0 pre-merge pipeline.

`--post-merge ` *(OPTIONAL)* : Run the L0 post-merge pipeline instead of the ordinary L0 pre-merge pipeline.

`--extra-stage "H100_PCIe-[Post-Merge]-1, xxx"` *(OPTIONAL)* : Run the ordinary L0 pre-merge pipeline and specified test stages. Examples: --extra-stage "H100_PCIe-[Post-Merge]-1, xxx".

### kill

`kill  `

Kill all running builds associated with pull request.

### skip

`skip --comment COMMENT `

Skip testing for latest commit on pull request. `--comment "Reason for skipping build/test"` is required. IMPORTANT NOTE: This is dangerous since lack of user care and validation can cause top of tree to break.

### reuse-pipeline

`reuse-pipeline `

Reuse a previous pipeline to validate current commit. This action will also kill all currently running builds associated with the pull request. IMPORTANT NOTE: This is dangerous since lack of user care and validation can cause top of tree to break.

</details>

Funatiq committed:
refactor: update speculative decoding buffer setup in DecoderState

- Consolidated speculative decoding buffer setup in single method.
- Adjusted calls to the new setup method in trtGptModelInflightBatching and gptDecoderBatchedTest to reflect changes in the API.

Signed-off-by: Robin Kobus <19427718+Funatiq@users.noreply.github.com>

Funatiq commented:
/bot run

tensorrt-cicd commented:
[PR_Github #8305](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/8305/) [ run ] triggered by Bot

Funatiq commented:
/bot run

tensorrt-cicd commented:
[PR_Github #8340](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/8340/) [ run ] triggered by Bot

tensorrt-cicd commented:
[PR_Github #8340](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/8340/) [ run ] completed with state `SUCCESS`
[/LLM/main/L0_MergeRequest_PR pipeline #6041](https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/6041/) completed with status: 'FAILURE'

Funatiq commented:
/bot run

tensorrt-cicd commented:
[PR_Github #8427](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/8427/) [ run ] completed with state `SUCCESS`
[/LLM/main/L0_MergeRequest_PR pipeline #6106](https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/6106/) completed with status: 'FAILURE'

Funatiq commented:
/bot run --disable-fail-fast

tensorrt-cicd commented:
[PR_Github #8474](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/8474/) [ run ] completed with state `SUCCESS`
[/LLM/main/L0_MergeRequest_PR pipeline #6139](https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/6139/) completed with status: 'FAILURE'

Funatiq commented:
/bot run --disable-multi-gpu-test

Funatiq commented:
/bot run --only-multi-gpu-test

tensorrt-cicd commented:
[PR_Github #8685](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/8685/) [ run ] completed with state `SUCCESS`
[/LLM/main/L0_MergeRequest_PR pipeline #6294 `(Partly Tested)`](https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/6294/) completed with status: 'FAILURE'

Funatiq commented:
/bot run --only-multi-gpu-test

DomBrown reviewed (approved):
LGTM

tensorrt-cicd commented:
[PR_Github #8840](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/8840/) [ run ] triggered by Bot

tensorrt-cicd commented:
[PR_Github #8840](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/8840/) [ run ] completed with state `SUCCESS`
[/LLM/main/L0_MergeRequest_PR pipeline #6427 `(Partly Tested)`](https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/6427/) completed with status: 'SUCCESS'

tensorrt-cicd commented:
[PR_Github #8881](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/8881/) [ run ] triggered by Bot

tensorrt-cicd commented:
[PR_Github #8881](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/8881/) [ run ] completed with state `SUCCESS`
[/LLM/main/L0_MergeRequest_PR pipeline #6467](https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/6467/) completed with status: 'SUCCESS'

Funatiq commented:
/bot run

tensorrt-cicd commented:
[PR_Github #8891](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/8891/) [ run ] completed with state `SUCCESS`
[/LLM/main/L0_MergeRequest_PR pipeline #6476](https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/6476/) completed with status: 'FAILURE'

Funatiq commented:
/bot run

tensorrt-cicd commented:
[PR_Github #8899](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/8899/) [ run ] completed with state `SUCCESS`
[/LLM/main/L0_MergeRequest_PR pipeline #6484](https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/6484/) completed with status: 'SUCCESS'

Funatiq committed:
chore: Add default constructors for decoder input/output

Signed-off-by: Robin Kobus <19427718+Funatiq@users.noreply.github.com>

Funatiq committed:
refactor: Modify DecoderState constructor and setup method

- Changed DecoderState to use a default constructor, removing the need for a DataType and BufferManager during instantiation.
- Updated the setup method to include DataType as a parameter, ensuring compatibility with the new constructor.
- Adjusted calls to DecoderState in trtGptModelInflightBatching, gptDecoderBatchedTest, and sampler.py to reflect the updated API.

Signed-off-by: Robin Kobus <19427718+Funatiq@users.noreply.github.com>

Funatiq committed:
fixup! refactor: Modify DecoderState constructor and setup method

Signed-off-by: Robin Kobus <19427718+Funatiq@users.noreply.github.com>

Funatiq commented:
/bot run

tensorrt-cicd commented:
[PR_Github #9721](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/9721/) [ run ] triggered by Bot

tensorrt-cicd commented:
[PR_Github #9721](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/9721/) [ run ] completed with state `SUCCESS`
[/LLM/main/L0_MergeRequest_PR pipeline #7156](https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/7156/) completed with status: 'FAILURE'

Funatiq commented:
/bot run

Funatiq commented:
/bot run

tensorrt-cicd commented:
[PR_Github #9819](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/9819/) [ run ] completed with state `SUCCESS`
[/LLM/main/L0_MergeRequest_PR pipeline #7245](https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/7245/) completed with status: 'FAILURE'

Funatiq commented:
/bot run

tensorrt-cicd commented:
[PR_Github #9859](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/9859/) [ run ] completed with state `SUCCESS`
[/LLM/main/L0_MergeRequest_PR pipeline #7272](https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/7272/) completed with status: 'SUCCESS'
Pipeline passed with automatic retried tests. Check the [rerun report](https://urm.nvidia.com/artifactory/sw-tensorrt-generic/llm-artifacts//LLM/main/L0_MergeRequest_PR/7272/test-results/rerun_report.html) for details.

Funatiq commented:
/bot run

tensorrt-cicd commented:
[PR_Github #10244](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/10244/) [ run ] triggered by Bot

tensorrt-cicd commented:
[PR_Github #10244](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/10244/) [ run ] completed with state `SUCCESS`
[/LLM/main/L0_MergeRequest_PR pipeline #7571](https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/7571/) completed with status: 'SUCCESS'
Pipeline passed with automatic retried tests. Check the [rerun report](https://urm.nvidia.com/artifactory/sw-tensorrt-generic/llm-artifacts//LLM/main/L0_MergeRequest_PR/7571/test-results/rerun_report.html) for details.
